Metalama.Framework.Engine/Pipeline/SourceTransformer.cs

```diff
@@ -54,9 +54,9 @@ public sealed class SourceTransformer : ISourceTransformerWithServices
                 DotNetSdkDirectory = dotNetSdkDirectory
             };
 
-            BackstageServiceFactoryInitializer.Initialize( backstageOptions );
-
-            var serviceProvider = BackstageServiceFactory.ServiceProvider;
+            // We don't use BackstageServiceFactory.ServiceProvider here, because it's lifetime goes over the lifetime of a source transformer,
+            // and we manage disposal of services at the end of the source transformer's lifetime.
+            var serviceProvider = BackstageServiceFactoryInitializer.CreateInitialized( backstageOptions );
```

Comment:
There are other usages of `BackstageServiceFactory.ServiceProvider` in the solution. Do these need to be reviewed?

Reply:
I've reviewed those.